js-api-doc/legacy/function.d.ts

```diff
@@ -1,94 +1,661 @@
+ */
 export type LegacyAsyncFunction = (
-  this: PluginThis,
-  ...args: [...LegacyValue[], (type: LegacyValue) => void]
+  this: LegacyPluginThis,
+  ...args: [...LegacyValue[], (result: LegacyValue) => void]
 ) => void;
 
+/**
+ * A callback that implements a custom Sass function. For [[renderSync]], this
+ * must be a [[LegacySyncFunction]] which returns its result directly; for
+ * [[render]], it may be either a [[LegacySyncFunction]] or a
+ * [[LegacyAsyncFunction]] which calls a callback with its result.
+ *
+ * See [[LegacySharedOptions.functions]] for more details.
+ */
 export type LegacyFunction<sync extends 'sync' | 'async'> = sync extends 'async'
   ? LegacySyncFunction | LegacyAsyncFunction
   : LegacySyncFunction;
 
+/**
+ * A type representing all the possible values that may be passed to or returned
+ * from a [[LegacyFunction]].
+ */
 export type LegacyValue =
   | types.Null
   | types.Number
   | types.String
   | types.Boolean
-  | types.HslColor
-  | types.RgbColor
+  | types.Color
   | types.List
   | types.Map;
 
+/** The namespace for value types used in the legacy function API. */
 export namespace types {
+  /**
+   * The class for Sass's singleton [`null`
+   * value](https://sass-lang.com/documentation/values/null). The value itself
+   * can be accessed through the [[NULL]] field.
+   */
   export class Null {
-    static NULL: Null;
+    /** Sass's singleton `null` value. */
+    static readonly NULL: Null;
   }
 
+  /**
+   * Sass's [number type](https://sass-lang.com/documentation/values/numbers).
+   */
   export class Number {
+    /**
+     * @param value - The numeric value of the number.
+     *
+     * @param unit - If passed, the number's unit.
+     *
+     * Complex units can be represented as
+     * `<unit>*<unit>*.../<unit>*<unit>*...`, with numerator units on the
+     * left-hand side of the `/` and denominator units on the right. A number
+     * with only numerator units may omit the `/` and the units after it, and a
+     * number with only denominator units may be represented
+     * with no units before the `/`.
+     *
+     * @example
+     *
+     * ```scss
+     * new sass.types.Number(0.5); // == 0.5
+     * new sass.types.Number(10, "px"); // == 10px
+     * new sass.types.Number(10, "px*px"); // == 10px * 1px
+     * new sass.types.Number(10, "px/s"); // == math.div(10px, 1s)
+     * new sass.types.Number(10, "px*px/s*s"); // == 10px * math.div(math.div(1px, 1s), 1s)
+     * ```
+     */
     constructor(value: number, unit?: string);
+
+    /**
+     * Returns the value of the number, ignoring units.
+     *
+     * <div class="sl-c-callout sl-c-callout--warning"><h3>⚠ Heads up!</h3>
+     *
+     * This means that `96px` and `1in` will return different values, even
+     * though they represent the same length.
+     *
+     * </div>
+     *
+     * @example
+     *
+     * ```js
+     * var number = new sass.types.Number(10, "px");
+     * console.log(number.getValue()); // 10
```

Comment:
Nit (here and below): Other examples (color/string) are using `value.getValue(); // 'val'` instead of `console.log(value.getValue()); // 'val'`. Consider aligning onto one style.